domain/src/test/java/org/oppia/android/domain/devoptions/ModifyLessonProgressControllerTest.kt

```diff
@@ -308,6 +315,65 @@ class ModifyLessonProgressControllerTest {
     }
   }
 
+  @Test
+  fun markFirstAndFractionsTopicsCompleted_TopicsAreCompleted() {
+    modifyLessonProgressController.markMultipleTopicsCompleted(
+      profileId,
+      listOf(TEST_TOPIC_ID_0, FRACTIONS_TOPIC_ID)
+    )
+    val allTopics = retrieveAllTopics()
+    val firstTopic = allTopics[0]
+    val fractionsTopic = allTopics[2]
+    assertThat(firstTopic.topicId).isEqualTo(TEST_TOPIC_ID_0)
+    assertThat(firstTopic.storyList[0].chapterList[0].chapterPlayState)
+      .isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(firstTopic.storyList[0].chapterList[1].chapterPlayState)
+      .isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(fractionsTopic.topicId).isEqualTo(FRACTIONS_TOPIC_ID)
+    assertThat(fractionsTopic.storyList[0].chapterList[0].chapterPlayState)
+      .isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(fractionsTopic.storyList[0].chapterList[1].chapterPlayState)
+      .isEqualTo(ChapterPlayState.COMPLETED)
+  }
+
+  @Test
+  fun markFirstAndRatios2StoriesCompleted_StoriesAreCompleted() {
+    modifyLessonProgressController.markMultipleStoriesCompleted(
+      profileId,
+      mapOf(TEST_STORY_ID_0 to TEST_TOPIC_ID_0, RATIOS_STORY_ID_1 to RATIOS_TOPIC_ID)
+    )
+    val allStories = retrieveAllStories()
+    val firstStory = allStories[0]
+    val ratios2Story = allStories[4]
+    assertThat(firstStory.storyId).isEqualTo(TEST_STORY_ID_0)
+    assertThat(firstStory.chapterList[0].chapterPlayState).isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(firstStory.chapterList[1].chapterPlayState).isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(ratios2Story.storyId).isEqualTo(RATIOS_STORY_ID_1)
+    assertThat(ratios2Story.chapterList[0].chapterPlayState).isEqualTo(ChapterPlayState.COMPLETED)
+    assertThat(ratios2Story.chapterList[1].chapterPlayState).isEqualTo(ChapterPlayState.COMPLETED)
+  }
+
+  @Test
+  fun markPrototypeAndBothFractionsExplorationsCompleted_ExplorationsAreCompleted() {
```

Comment:
```suggestion
  fun markPrototypeAndBothFractionsExplorationsCompleted_allExplorationsAreCompleted() {
```

Reply:
Changed to `markPrototypeAndBothFractionsExplorationsCompleted_allThreeExplorationsAreCompleted`. There is some slight difference. PTAL. Thanks.